src/mptool.f90

```diff
@@ -132,7 +152,134 @@ real(mytype) function  pmax_mytype(var)
                                                     mpi_comm_world,ierr)
     !
   end function pmax_mytype
+
+  real(mytype) function  pmax_mytype_array(var)
```

Comment:
Please make the size of the output explicit so that it is easier to read / understand the function

Reply:
add a new optional parameter var_size